### PR TITLE
Add resnet-pytorch

### PR DESCRIPTION
**Pull request by Pramodh-G**
models with writeups.

### CODE REVIEW

**Comment by nirmal-suthar** [comment]
can you make folder `ImageClassification` and add your resnet folder inside it.

File: IMAGE-CLASSIFICATION/ResNet-PyTorch/main.py [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+import importlib
+import json
+from collections import OrderedDict
+import logging
+import argparse
+import numpy as np
+import random
+import time
+from eval import plot_accuracy_epoch,plot_loss_epoch,make_heat_map
+from tqdm.notebook import tqdm
```
Comment (nirmal-suthar) [inline IMAGE-CLASSIFICATION/ResNet-PyTorch/main.py:14]:
```suggestion
from tqdm import tqdm
```

File: IMAGE-CLASSIFICATION/ResNet-PyTorch/main.py [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+import torch.backends.cudnn
+import torchvision.utils
+import torch.nn.functional as F
+
+from dataloader import get_loader
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    # model config
+    parser.add_argument('--block_type', type=str, required=True)
```
Comment (nirmal-suthar) [inline IMAGE-CLASSIFICATION/ResNet-PyTorch/main.py]: add default to block_type

File: IMAGE-CLASSIFICATION/ResNet-PyTorch/main.py [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+    parser.add_argument('--depth', type=int, required=True)
```
Comment (nirmal-suthar) [inline IMAGE-CLASSIFICATION/ResNet-PyTorch/main.py]: add default to depth

File: IMAGE-CLASSIFICATION/ResNet-PyTorch/resnet.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
+# A short summary and implementation on the ResNet model(CIFAR-10 Dataset)
+
```
Comment (nirmal-suthar) [inline IMAGE-CLASSIFICATION/ResNet-PyTorch/resnet.md]: rename resnet.md file to README.md and instruction to run the model  something like `python3 main.py --block_type "basic" ` and give reference to the paper

File: IMAGE-CLASSIFICATION/ResNet-PyTorch/main.py [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+
+model.to(device)
+criterion = nn.CrossEntropyLoss()
+train_loader,test_loader = get_loader(config['optim_config']['batch_size'],config['run_config']['num_workers'])
+
+train_loss,test_loss,train_accuracy,test_accuracy = train(model,config['optim_config']['epochs'],train_loader,test_loader,device,criterion,optimizer,scheduler)
+
+plot_loss_epoch(train_loss,test_loss)
+plot_accuracy_epoch(train_accuracy,test_accuracy)
+make_heat_map(model,test_loader,device)
```
Comment (nirmal-suthar) [inline IMAGE-CLASSIFICATION/ResNet-PyTorch/main.py]: instead of plotting, save the plot as file.

**Comment by nirmal-suthar** [comment]
Also, it would be great to seperate model code in separate file named `models.py`

File: IMAGE-CLASSIFICATION/ResNet-PyTorch/main.py (hunk 5 above)
Comment (Pramodh-G) [reply IMAGE-CLASSIFICATION/ResNet-PyTorch/main.py]: did both plotting and saving plot to the file.
add models.py
add reference to paper and how to run code on README.md

**Pull request merged**